clang/lib/Driver/Driver.cpp

```diff
@@ -5970,7 +5970,10 @@ bool clang::driver::isObjectFile(std::string FileName) {
   if (!llvm::sys::path::has_extension(FileName))
     // Any file with no extension should be considered an Object. Take into
     // account -lsomelib library filenames.
-    return FileName.rfind("-l", 0) != 0;
+    // When -lstdc++ is passed in certain cases it's transformed into an driver 
+    // option.   
+    return (FileName.rfind("-l", 0) != 0 && 
+            FileName != "-Z-reserved-lib-stdc++"); 
```

Comment:
Could you please attempt factoring this out into (another :)) separate function, e.g. `isLibraryCall`?
We could then add new cases to this new function's scope without having to stack conditions in `isObjectFile()`.

Reply:
Sure, happy to do so! Will do it in my next commit, should I add a check for Z_reserved_lib_cckext as well while I am there?